core/src/main.cpp

```diff
@@ -360,13 +360,33 @@ constexpr auto default_jsonschema() -> std::string_view
                                 "cert",
                                 "hostname"
                             ]
+                        }},
+                        "client_server_negotiation": {{
+                            "type": "string"
+                        }},
+                        "encryption_algorithm": {{
+                            "type": "string"
+                        }},
+                        "encryption_key_size": {{
+                            "type": "integer"
+                        }},
+                        "encryption_hash_rounds": {{
+                            "type": "integer"
+                        }},
+                        "encryption_salt_size": {{
+                            "type": "integer"
                         }}
                     }},
                     "required": [
                         "client_server_policy",
                         "ca_certificate_file",
                         "dh_params_file",
-                        "verify_server"
+                        "verify_server",
+                        "client_server_negotiation",
+                        "encryption_algorithm",
+                        "encryption_key_size",
+                        "encryption_hash_rounds",
+                        "encryption_salt_size"
```

Comment:
like @alanking said, most of these are only required if client_server_policy is not `CS_NEG_REFUSE`.

Reply:
Correct. They are only required when the `tls` property is defined. That forces the admin to set those values accordingly.

Reply:
so... do we want to require them here in the schema?    doesn't seem quite right.

but like always, it's a balance between requiring, offering default examples, and the code handling a missing value and using a hard-coded default...

Reply:
If an admin doesn't plan on enabling TLS, they don't need to include the `tls` stanza at all. Once they include that property, the schema requires them to define all of the other properties.

If we relax that, it's easier for admins to get the TLS settings incorrect.

There's also the possibility of making the schema more complicated by changing the validation based on the value of `client_server_negotiation`. I don't think making the schema more complicated is worth it at this time.

Side note: Perhaps `dh_params_file` should be optional and the chain file property should be required. Hmm.

Reply:
i think that logic is okay/good.   yes, if they have a tls section, they should have all of it.  that's fine.

regarding dh_params... let's require it as good practice for now.  loosen if we get pushback.    required value can always be empty string...

Reply:
What are your thoughts on the chain file property becoming required?

Asking based on this: https://docs.irods.org/4.3.1/plugins/pluggable_authentication/#server-ssl-setup

See the first sentence of the section **Create the certificate chain file**.

Reply:
chain file is for the irods server...

compare with the new password booth (just a client)
https://github.com/irods/irods_client_password_booth/blob/main/app.config.template

Reply:
that suggests no dh_params either

Reply:
Cool cool. Will investigate the removal of the dh_params and chain file properties and report what I find.

Reply:
Removed properties.